internal/provider/kubernetes/controller.go

```diff
@@ -999,9 +995,17 @@ func (r *gatewayAPIReconciler) processGateways(ctx context.Context, managedGC *g
 				return err
 			}
 		}
+
 		// Discard Status to reduce memory consumption in watchable
 		// It will be recomputed by the gateway-api layer
 		gtw.Status = gwapiv1.GatewayStatus{}
+
+		if err := r.processGatewayParamsRef(ctx, &gtw, resourceMap, resourceTree); err != nil {
+			// Update the Gateway status to not accepted if there is an error processing the parametersRef
+			status.UpdateGatewayStatusNotAccepted(&gtw, gwapiv1.GatewayReasonInvalidParameters, err.Error())
+			r.log.Error(err, "failed to process infrastructure.parametersRef for gateway", "namespace", gtw.Namespace, "name", gtw.Name)
+		}
```

Comment:
I would prefer to add a `continue` here, otherwise the gateway instance will be provisoned?

Reply:
This is tricky because:
* If we skip adding this to the `resourceTree`, the Gateway status won't be updated.
* If we update the status directly in the provider controller, it introduces a potential race condition: both the provider controller goroutine and the Gateway API translator goroutine will attempte to write to resources.GatewayStatus.

Reply:
Skipped Gateways with invalid EnvoyProxy in https://github.com/envoyproxy/gateway/pull/5117/commits/8f51c9a6b131c54738cea8a4af9a60c942b439af